ffi/blitbuffer.lua

```diff
@@ -1912,4 +1911,28 @@ BB.TYPE_BBRGB16 = TYPE_BBRGB16
 BB.TYPE_BBRGB24 = TYPE_BBRGB24
 BB.TYPE_BBRGB32 = TYPE_BBRGB32
 
+BB.has_cblitbuffer = false
+if not os.getenv("KO_NO_CBB") then
+    -- Load C blit buffer, we'll decide whether to use it later on
+    BB.has_cblitbuffer, cblitbuffer = pcall(ffi.load, "blitbuffer")
+end
+
+-- Set the actual enable/disable CBB flag. Returns the flag of whether it is (actually) enabled.
+function BB:enableCBB(ison)
+    local old = use_cblitbuffer
+    use_cblitbuffer = ison and self.has_cblitbuffer
+    if old ~= use_cblitbuffer then
+        -- NOTE: This works-around a number of corner-cases which may end up with LuaJIT's optimizer blacklisting this very codepath,
+        --       which'd obviously *murder* performance (to the effect of a soft-lock, essentially).
+        --       c.f., #4137, #4752, #4782
```

Comment:
While we're unlikely to hit 4000 anytime soon (if ever), I think it's nevertheless clearer/safer to specify.

```suggestion
        --       c.f., koreader/koreader#4137, koreader/koreader#4752, koreader/koreader#4782
```

```suggestion
        --       c.f., koreader#4137, koreader#4752, koreader#4782
```